### PR TITLE
[hash]: Add ECMP/LAG hash algorithm to OA

### PR DESCRIPTION
**Pull request by nazariig**
Signed-off-by: Nazarii Hnydyn [nazariig@nvidia.com](mailto:nazariig@nvidia.com)

**DEPENDS:**
1. https://github.com/sonic-net/sonic-sairedis/pull/1316

**HLD:** https://github.com/sonic-net/SONiC/pull/1501

**What I did**
* Implemented Generic Hash feature

**Why I did it**
* Implementation is done according to the Generic Hash HLD

**How I verified it**
1. Run Generic Hash VS UTs

**Details if related**
* N/A

### CODE REVIEW

**Comment by nazariig** [comment]
@stepanblyschak FYI

**Review by stepanblyschak — APPROVED** [review]

**Comment by liat-grozovik** [comment]
@prsunny kindly reminder to review or to assign someone on behalf of you.
this is for 202311 and should get in so better to complete before branch out.
in any case, if you wish to have another reviewer from Nvidia to approve the PRs, please let me know and i will assign.

**Comment by nazariig** [comment]
@prsunny just a kind reminder

**Comment by nazariig** [comment]
@prsunny just a kind reminder

**Comment by nazariig** [comment]
@prsunny just a kind reminder

File: orchagent/switch/switch_capabilities.cpp [hunk 1]
```diff
@@ -139,14 +206,49 @@ bool SwitchCapabilities::validateSwitchHashFieldCap(const std::set<sai_native_ha
...
+
+    if (obj.haSet.empty())
+    {
+        SWSS_LOG_ERROR("Failed to validate hash algorithm: no hash algorithm capabilities");
+        return false;
+    }
+
+    if (obj.haSet.count(haValue) == 0)
+    {
+        SWSS_LOG_ERROR("Failed to validate hash algorithm: value(%s) is not supported", toStr(haValue).c_str());
```
Comment (prsunny) [inline orchagent/switch/switch_capabilities.cpp:245]: Same here. Please change to WARN

**Review by prsunny — APPROVED** [review]
lgtm, minor comment

**Comment by liushilongbuaa** [comment]
Already in 202311.
No action is needed.

**Pull request merged**